Src/AutoNSubstituteUnitTest/AutoConfiguredFixtureIntegrationTest.cs

```diff
@@ -427,23 +427,5 @@ public void InterfaceShadowedMethodsReturnValueFromFixture()
             Assert.Same(frozenString, (result as IInterfaceWithShadowedMethod).Method(0));
             // Teardown
         }
-
-        [Fact]
-        public void InterfacesImplementingIEnumerableReturnFiniteSequence()
```

Comment:
While I understand that, under the hood, this test ends up invoking `GetEnumerator`, that's doesn't look like it's the purpose of this test. What I'm trying to say is that this unit test seems to operate on another abstraction level than the tests that explicitly go for `IEnumerator<T>`.

It looks to me like this is a slightly more behavioural test that's not particularly concerned with the implementation details of `GetEnumerator`. Perhaps this test was added to reproduce a defect. Isn't it worth to keep such a test around as a regression test?